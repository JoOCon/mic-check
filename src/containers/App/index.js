import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, withRouter} from 'react-router-dom';
import PropTypes from 'prop-types';
import { 
  getLocalStorageUsers, 
  getLocalStorageRentals, 
  getLocalStorageOrders 
} from '../../actions';
import { Header } from '../../components/Header';
import Login from '../Login';
import SignUp from '../SignUp';
import RentalForm from '../RentalForm';
import RentalsContainer from '../RentalsContainer';
import UserAccountDisplay from '../UserAccountDisplay';
import RentalConfirmation from '../RentalConfirmation';
import './styles.css';

export class App extends Component {
  componentDidMount() {
    const {
      getLocalStorageUsers,
      getLocalStorageRentals,
      getLocalStorageOrders
    } = this.props;
    const users = JSON.parse(localStorage.getItem('AllUsers'));
    const rentals = JSON.parse(localStorage.getItem('AllRentals'));
    const orders = JSON.parse(localStorage.getItem('AllCompletedOrders'));
    getLocalStorageUsers(users);
    getLocalStorageRentals(rentals);
    getLocalStorageOrders(orders);
  }

  render() {
    const { userName } = this.props.activeUser;
    return (
      <div className="App">
        <Route path='/' component={Header} />
        <Route exact path='/Login' component={Login} />
        <Route exact path='/SignUp' component={SignUp} />
        <Route exact path={`/${userName}/AddRental`} component={RentalForm} />
        <Route exact path='/' component={RentalsContainer} />
        <Route exact path={`/${userName}`} component={RentalsContainer} />
        <Route exact path={`/${userName}/Orders`} component={UserAccountDisplay} />
        <Route exact path={`/${userName}/Confirmation`} component={RentalConfirmation} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  activeUser: state.activeUser
});

export const mapDispatchToProps = (dispatch) => ({
  getLocalStorageUsers: (users) => dispatch(getLocalStorageUsers(users)),
  getLocalStorageRentals: (rentals) => dispatch(getLocalStorageRentals(rentals)),
  getLocalStorageOrders: (orders) => dispatch(getLocalStorageOrders(orders))
});

App.propTypes = {
  activeUser: PropTypes.object,
  getLocalStorageUsers: PropTypes.func,
  getLocalStorageRentals: PropTypes.func,
  getLocalStorageOrders: PropTypes.func
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
