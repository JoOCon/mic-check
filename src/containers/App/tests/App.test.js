import React from 'react';
import ReactDOM from 'react-dom';
import App from '../';

describe('App Tests', () => {
  it('should render App with the correct props', () => {
    expect(true).toEqual(true)
  });
})
