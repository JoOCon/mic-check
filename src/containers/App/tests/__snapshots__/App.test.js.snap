// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders the App with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    activeUser={
        Object {
            "email": "THanks@hanks.com",
            "location": "20 McAllister St, San Francisco, CA",
            "password": "password",
            "userName": "Tom Hanks",
          }
    }
    getLocalStorageOrders={[Function]}
    getLocalStorageRentals={[Function]}
    getLocalStorageUsers={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/Login"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/SignUp"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/Tom Hanks/AddRental"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/Tom Hanks"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/Tom Hanks/Orders"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/Tom Hanks/Confirmation"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/Login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/SignUp",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/Tom Hanks/AddRental",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/Tom Hanks",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/Tom Hanks/Orders",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/Tom Hanks/Confirmation",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/Login"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/SignUp"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/Tom Hanks/AddRental"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/Tom Hanks"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/Tom Hanks/Orders"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/Tom Hanks/Confirmation"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/Login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/SignUp",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/Tom Hanks/AddRental",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/Tom Hanks",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/Tom Hanks/Orders",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/Tom Hanks/Confirmation",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
